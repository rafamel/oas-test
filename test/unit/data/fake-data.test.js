import fakeData from '../../../src/data/fake-data';
import operations from './operations';
import jsf from 'json-schema-faker';

// Set randomness generator for faker as constant
jsf.option({ random: () => 0.5 });

describe(`Single Mime Type`, () => {
  test(`No parameters`, () => {
    const fake = fakeData(operations.singleMime.noParams);
    expect(fake).toMatchSnapshot();
  });
  test(`No request body`, () => {
    const fake = fakeData(operations.singleMime.noBody);
    expect(fake).toMatchSnapshot();
  });
  test(`Parameters and request body`, () => {
    const fake = fakeData(operations.singleMime.all);
    expect(fake).toMatchSnapshot();
  });
});

describe(`Multiple Mime Type`, () => {
  test(`No parameters`, () => {
    const fake = fakeData(operations.multipleMimes.noParams);
    expect(fake).toMatchSnapshot();
  });
  test(`No request body`, () => {
    const fake = fakeData(operations.multipleMimes.noBody);
    expect(fake).toMatchSnapshot();
  });
  test(`Parameters and request body`, () => {
    const fake = fakeData(operations.multipleMimes.all);
    expect(fake).toMatchSnapshot();
  });
});
