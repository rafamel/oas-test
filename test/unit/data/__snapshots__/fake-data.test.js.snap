// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Mime Type No parameters 1`] = `
Object {
  "all": Object {
    "application/json": Object {
      "body": Object {
        "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
        "password": "FFFFFF FFFFFF",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {},
      "params": Object {},
      "query": Object {},
    },
    "application/xml": Object {
      "body": Object {
        "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
        "password": "FFFFFF FFFFFF",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {},
      "params": Object {},
      "query": Object {},
    },
  },
  "mimeTypes": Array [
    "application/json",
    "application/xml",
  ],
  "required": Object {
    "application/json": Object {
      "body": Object {
        "password": "FFFFFF FFFFFF",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {},
      "params": Object {},
      "query": Object {},
    },
    "application/xml": Object {
      "body": Object {
        "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {},
      "params": Object {},
      "query": Object {},
    },
  },
}
`;

exports[`Multiple Mime Type No request body 1`] = `
Object {
  "all": Object {
    "application/json": Object {
      "body": Object {},
      "cookies": Object {},
      "headers": Object {
        "some": 0,
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {
        "name": 0,
      },
    },
    "application/xml": Object {
      "body": Object {},
      "cookies": Object {},
      "headers": Object {
        "some": "Lorem anim ipsum",
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {
        "name": "Lorem anim ipsum",
      },
    },
  },
  "mimeTypes": Array [
    "application/xml",
    "application/json",
  ],
  "required": Object {
    "application/json": Object {
      "body": Object {},
      "cookies": Object {},
      "headers": Object {
        "some": 0,
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {},
    },
    "application/xml": Object {
      "body": Object {},
      "cookies": Object {},
      "headers": Object {
        "some": "Lorem anim ipsum",
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {},
    },
  },
}
`;

exports[`Multiple Mime Type Parameters and request body 1`] = `
Object {
  "all": Object {
    "application/json": Object {
      "body": Object {
        "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
        "password": "FFFFFF FFFFFF",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {
        "some": 0,
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {
        "name": 0,
      },
    },
    "application/xml": Object {
      "body": Object {
        "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
        "password": "FFFFFF FFFFFF",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {
        "some": "Lorem anim ipsum",
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {
        "name": "Lorem anim ipsum",
      },
    },
  },
  "mimeTypes": Array [
    "application/json",
    "application/xml",
  ],
  "required": Object {
    "application/json": Object {
      "body": Object {
        "password": "FFFFFF FFFFFF",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {
        "some": 0,
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {},
    },
    "application/xml": Object {
      "body": Object {
        "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
        "username": "FFFFFF",
      },
      "cookies": Object {},
      "headers": Object {
        "some": "Lorem anim ipsum",
      },
      "params": Object {
        "id": 50000000,
      },
      "query": Object {},
    },
  },
}
`;

exports[`Single Mime Type No parameters 1`] = `
Object {
  "all": Object {
    "body": Object {
      "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
      "password": "FFFFFF FFFFFF",
      "username": "FFFFFF",
    },
    "cookies": Object {},
    "headers": Object {},
    "params": Object {},
    "query": Object {},
  },
  "mimeTypes": Array [],
  "required": Object {
    "body": Object {
      "password": "FFFFFF FFFFFF",
      "username": "FFFFFF",
    },
    "cookies": Object {},
    "headers": Object {},
    "params": Object {},
    "query": Object {},
  },
}
`;

exports[`Single Mime Type No request body 1`] = `
Object {
  "all": Object {
    "body": Object {},
    "cookies": Object {
      "aCookie": "Lorem anim ipsum",
    },
    "headers": Object {
      "some": "Lorem anim ipsum",
    },
    "params": Object {
      "id": 50000000,
    },
    "query": Object {
      "name": 50000000,
    },
  },
  "mimeTypes": Array [],
  "required": Object {
    "body": Object {},
    "cookies": Object {},
    "headers": Object {
      "some": "Lorem anim ipsum",
    },
    "params": Object {
      "id": 50000000,
    },
    "query": Object {},
  },
}
`;

exports[`Single Mime Type Parameters and request body 1`] = `
Object {
  "all": Object {
    "body": Object {
      "email": "FFFFFFFFF@AAAAAAAAAAAAAAAAA.nnn",
      "password": "FFFFFF FFFFFF",
      "username": "FFFFFF",
    },
    "cookies": Object {
      "aCookie": "Lorem anim ipsum",
    },
    "headers": Object {
      "some": "Lorem anim ipsum",
    },
    "params": Object {
      "id": 50000000,
    },
    "query": Object {
      "name": 50000000,
    },
  },
  "mimeTypes": Array [],
  "required": Object {
    "body": Object {
      "password": "FFFFFF FFFFFF",
      "username": "FFFFFF",
    },
    "cookies": Object {},
    "headers": Object {
      "some": "Lorem anim ipsum",
    },
    "params": Object {
      "id": 50000000,
    },
    "query": Object {},
  },
}
`;
